src/targets/__tests__/awsLambda.test.ts

```diff
@@ -0,0 +1,261 @@
+    try {
+      const multiplePackagesFound = await awsTarget.publish(
+        'version',
+        'revision'
+      );
+      expect(multiplePackagesFound).toBe(undefined);
+    } catch (error) {
+      expect(error instanceof Error).toBe(true);
+      const multiplePackagesPattern = /multiple packages/;
+      expect(multiplePackagesPattern.test(error.message)).toBe(true);
+    }
+  });
+
+  const singleArtifactsForRevision = jest
+    .fn()
+    .mockImplementation(function(input: string) {
+      return [input];
+    });
+
+  const downloadArtifactMock = jest
+    .fn()
+    .mockImplementation(function(input: string) {
+      return input;
+    });
+
+  const publishAwsLayerMock = jest.fn().mockImplementation(function() {
+    return {
+      Version: 1,
+      LayerVersionArn: 'layer:version:arn:test',
+    };
+  });
+  const publishAwsLayerMockUndefinedVersion = jest
+    .fn()
+    .mockImplementation(function() {
+      return {
+        Version: undefined,
+        LayerVersionArn: 'layer:version:arn:test',
+      };
+    });
+
+  const addLayerPermissionsMock = jest.fn().mockImplementation(function() {
+    // Do nothing
+  });
+
+  test('success on publishing', async () => {
+    const awsTarget = getAwsLambdaTarget();
+    setTestingProjectConfig(awsTarget);
+    awsTarget.getAwsRegions = getAwsRegionsMock.bind(AwsLambdaLayerTarget);
+    awsTarget.getArtifactsForRevision = singleArtifactsForRevision.bind(
+      AwsLambdaLayerTarget
+    );
+    awsTarget.artifactProvider.downloadArtifact = downloadArtifactMock.bind(
+      awsTarget
+    );
+    awsTarget.publishAwsLayer = publishAwsLayerMock.bind(AwsLambdaLayerTarget);
+    awsTarget.addAwsLayerPermissions = addLayerPermissionsMock.bind(
+      AwsLambdaLayerTarget
+    );
+    await awsTarget.publish('', '');
+    expect(singleArtifactsForRevision).toBeCalledWith('', {
+      includeNames: undefined,
+    });
+    expect(downloadArtifactMock).toBeCalledWith('');
+  });
+
+  test('error on layer version', async () => {
+    try {
+      const awsTarget = getAwsLambdaTarget();
+      setTestingProjectConfig(awsTarget);
+      awsTarget.getAwsRegions = getAwsRegionsMock.bind(AwsLambdaLayerTarget);
+      awsTarget.getArtifactsForRevision = singleArtifactsForRevision.bind(
+        AwsLambdaLayerTarget
+      );
+      awsTarget.artifactProvider.downloadArtifact = downloadArtifactMock.bind(
+        awsTarget
+      );
+      awsTarget.publishAwsLayer = publishAwsLayerMockUndefinedVersion.bind(
+        AwsLambdaLayerTarget
+      );
+      const publishedLayerVersion = await awsTarget.publish('', '');
+      // `publish` should report an error. When it's not dry run, the error is
+      // thrown; when it's on dry run, the error is logged and `undefined` is
+      // returned. Thus, both alternatives have been considered.
+      expect(publishedLayerVersion).toBe(undefined);
+    } catch (error) {
+      expect(error instanceof Error).toBe(true);
```

Comment:
To confirm: wouldn't this always be true?

If you want to confirm that a function throws an error, please consider using https://jestjs.io/docs/en/expect.html#tothrowerror

Reply:
The code inside `catch`, yes. `publish` will only throw if it's not running in dry-run mode; if it is, it doesn't throw anything and returns `undefined`. So, the approach is to run `publish` and expect to get `undefined`; if it throws an error before that, catch it. The method `toThrow`, afaik, requires to make the function call inside the `expect`, passing the test when the dry-run mode is off but failing when it's on.